# HG changeset patch
# User Rik <rik@octave.org>
# Date 1546997349 28800
#      Tue Jan 08 17:29:09 2019 -0800
# Branch stable
# Node ID dfc8d79cc28bc551fd12d890e303d4dd985eba75
# Parent  cfe28edab48b094099a56deafb0185c11bebf693
CDiagMatrix.cc: Fix static analyzer detected issues (bug #55347).

* CDiagMatrix.cc (ComplexDiagMatrix::row (octave_idx_type)): Delete useless
check on "r > c".
* CDiagMatrix.cc (ComplexDiagMatrix::row (char *s)): Get first character
with "s[0]" rather than "*s".
* CDiagMatrix.cc (ComplexDiagMatrix::column (octave_idx_type)): Delete useless
check on "r < c".
* CDiagMatrix.cc (ComplexDiagMatrix::column (char *s)): Get first character
with "s[0]" rather than "*s".

diff --git a/liboctave/array/CDiagMatrix.cc b/liboctave/array/CDiagMatrix.cc
--- a/liboctave/array/CDiagMatrix.cc
+++ b/liboctave/array/CDiagMatrix.cc
@@ -242,29 +242,29 @@ ComplexRowVector
 ComplexDiagMatrix::row (octave_idx_type i) const
 {
   octave_idx_type r = rows ();
   octave_idx_type c = cols ();
   if (i < 0 || i >= r)
     (*current_liboctave_error_handler) ("invalid row selection");
 
   ComplexRowVector retval (c, 0.0);
-  if (r <= c || (r > c && i < c))
+  if (r <= c || i < c)
     retval.elem (i) = elem (i, i);
 
   return retval;
 }
 
 ComplexRowVector
 ComplexDiagMatrix::row (char *s) const
 {
   if (! s)
     (*current_liboctave_error_handler) ("invalid row selection");
 
-  char c = *s;
+  char c = s[0];
   if (c == 'f' || c == 'F')
     return row (static_cast<octave_idx_type> (0));
   else if (c == 'l' || c == 'L')
     return row (rows () - 1);
   else
     (*current_liboctave_error_handler) ("invalid row selection");
 }
 
@@ -272,29 +272,29 @@ ComplexColumnVector
 ComplexDiagMatrix::column (octave_idx_type i) const
 {
   octave_idx_type r = rows ();
   octave_idx_type c = cols ();
   if (i < 0 || i >= c)
     (*current_liboctave_error_handler) ("invalid column selection");
 
   ComplexColumnVector retval (r, 0.0);
-  if (r >= c || (r < c && i < r))
+  if (r >= c || i < r)
     retval.elem (i) = elem (i, i);
 
   return retval;
 }
 
 ComplexColumnVector
 ComplexDiagMatrix::column (char *s) const
 {
   if (! s)
     (*current_liboctave_error_handler) ("invalid column selection");
 
-  char c = *s;
+  char c = s[0];
   if (c == 'f' || c == 'F')
     return column (static_cast<octave_idx_type> (0));
   else if (c == 'l' || c == 'L')
     return column (cols () - 1);
   else
     (*current_liboctave_error_handler) ("invalid column selection");
 }
 
